upup/pkg/kutil/node_api_adapter.go

```diff
@@ -0,0 +1,183 @@
+
+}
+
+// WaitForNodeToBeReady returns whether node name is ready within timeout.
+func (nodeAA *NodeAPIAdapter) WaitForNodeToBeReady(name string) bool {
+	return nodeAA.WaitForNodeToBe(name, api.NodeReady, true)
+}
+
+// WaitForNodeToBeNotReady returns whether node name is not ready (i.e. the
+// readiness condition is anything but ready, e.g false or unknown) within
+// timeout.
+func (nodeAA *NodeAPIAdapter) WaitForNodeToBeNotReady(name string) bool {
+	return nodeAA.WaitForNodeToBe(name, api.NodeReady, false)
+}
+
+// WaitForNodeToBe returns whether node "name's" condition state matches wantTrue
+// within timeout. If wantTrue is true, it will ensure the node condition status
+// is ConditionTrue; if it's false, it ensures the node condition is in any state
+// other than ConditionTrue (e.g. not true or unknown).
+func (nodeAA *NodeAPIAdapter) WaitForNodeToBe(name string, conditionType api.NodeConditionType, wantTrue bool) bool {
+	glog.V(4).Infof("Waiting up to %v for node %s condition %s to be %t", nodeAA.timeout, name, conditionType, wantTrue)
+	for start := time.Now(); time.Since(start) < nodeAA.timeout; time.Sleep(Poll) {
+		node, err := nodeAA.client.Nodes().Get(name)
+
+		// FIXME this is not erroring on 500's for instance.  We will keep looping
+		if err != nil {
+			glog.V(4).Infof("Couldn't get node %s", name)
+			continue
+		}
+
+		if IsNodeConditionSetAsExpected(node, conditionType, wantTrue) {
+			return true
+		}
+	}
+	glog.V(4).Infof("Node %s didn't reach desired %s condition status (%t) within %v", name, conditionType, wantTrue, nodeAA.timeout)
+	return false
+}
+
+
+func IsNodeConditionSetAsExpectedSilent(node *api.Node, conditionType api.NodeConditionType, wantTrue bool) bool {
+	return isNodeConditionSetAsExpected(node, conditionType, wantTrue, true)
+}
+
+func IsNodeConditionUnset(node *api.Node, conditionType api.NodeConditionType) bool {
+	for _, cond := range node.Status.Conditions {
+		if cond.Type == conditionType {
+			return false
+		}
+	}
+	return true
+}
+func FilterNodes(nodeList *api.NodeList, fn func(node api.Node) bool) {
+	var l []api.Node
+
+	for _, node := range nodeList.Items {
+		if fn(node) {
+			l = append(l, node)
+		}
+	}
+	nodeList.Items = l
+}
+
+func IsNodeConditionSetAsExpected(node *api.Node, conditionType api.NodeConditionType, wantTrue bool) bool {
+	return isNodeConditionSetAsExpected(node, conditionType, wantTrue, false)
+}
+
+// waitListSchedulableNodes is a wrapper around listing nodes supporting retries.
+func (nodeAA *NodeAPIAdapter) waitListSchedulableNodes() (nodes *api.NodeList, err error) {
+
+	if wait.PollImmediate(Poll, SingleCallTimeout, func() (bool, error) {
+		nodes, err = nodeAA.client.Nodes().List(api.ListOptions{FieldSelector: fields.Set{
+			"spec.unschedulable": "false",
+		}.AsSelector()})
+		if err != nil {
+			// error logging TODO
+			return false, err
+		}
+		return err == nil, nil
+	}) != nil {
+		// TODO logging
+		return nil, err
+	}
+	return nodes, nil
```

Comment:
I would just `return nodes, err` so that we ensure that no error is overlooked or forgotten about. Every caller should **all** check for an error and want to always reveal the true culprit.